wme_junctionangle.user.js

```diff
@@ -502,9 +529,28 @@ function run_ja() {
 					return ja_routing_type.EXIT;
 				}
 			}
+			
+			//wlodek76: FIXING KEEP LEFT/RIGHT regarding to left most segment
+			//WIKI WAZE: When there are more than two segments less than 45.04°, only the left most segment will be KEEP LEFT, all the rest will be KEEP RIGHT
+			
+			//wlodek76: KEEP LEFT/RIGHT overlapping case
+			//WIKI WAZE: If the left most segment is overlapping another segment, it will also be KEEP RIGHT.
+			if (mostleftangle!=null && mostleftangle2!=null) {
+				var overlapped_angle = Math.abs(mostleftangle[0] - mostleftangle2[0]);
+				
+				// If two top most left angles are close < 2 degree they are overlapped.
+				// Method of recognizing overlapped segment by server is unknown for me yet, I took this from WME Validator information about this.
+				// TODO: verify overlapping check on the side of routing server.
+				if (overlapped_angle < 2.0) mostleftangle = null;
+			}
+			
+			if (mostleftangle != null && mostleftangle[1] == s_out_id) {

```

Comment:
Another cases in Tokyo (LHT), this time keeps are on the left side of s_in.
Temp angle is 6 15 degree.

JAI patch and LiveMap shows the same keep left, keep right.

https://www.waze.com/pl/editor/?env=row&lon=139.78182&lat=35.63337&layers=388&zoom=8&segments=251168699
![image](https://cloud.githubusercontent.com/assets/7284315/7219100/156e96ba-e690-11e4-8d13-4b480c2304cd.png)